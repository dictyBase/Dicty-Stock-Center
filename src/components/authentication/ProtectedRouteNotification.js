// @flow
import React from "react"
import { SnackbarContent } from "material-ui/Snackbar"
import styled from "styled-components"

// change background color of snackbar
const SnackbarStyle = styled(SnackbarContent)`
  && {
    background: #a94442;
  }
`

type Props = {
  /** The error message to display */
  error: string
}

const ProtectedRouteNotification = (props: Props) => {
  return (
    <center>
      <SnackbarStyle message={props.error} />
      <br />
    </center>
  )
}

export default ProtectedRouteNotification
