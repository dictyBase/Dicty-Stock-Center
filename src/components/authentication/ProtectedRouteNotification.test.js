import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import ProtectedRouteNotification from "./ProtectedRouteNotification"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<ProtectedRouteNotification />, div)
})

test("matching a snapshot of ProtectedRouteNotification", () => {
  const component = renderer.create(<ProtectedRouteNotification />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
