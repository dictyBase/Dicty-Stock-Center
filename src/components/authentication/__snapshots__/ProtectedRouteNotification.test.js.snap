// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of ProtectedRouteNotification 1`] = `
.c0.c0 {
  background: #a94442;
}

<center>
  <div
    className="MuiTypography-root-31 MuiTypography-body1-40 MuiPaper-root-4 MuiPaper-elevation6-12 MuiSnackbarContent-root-1 c0"
    role="alertdialog"
  >
    <div
      className="MuiSnackbarContent-message-2"
    />
  </div>
  <br />
</center>
`;
